### PR TITLE
testing: unit test to cover valid XML, invalid structure

### PR DESCRIPTION
**Pull request by justinemmanuelmercado**

### CODE REVIEW

File: test/utils.ts [hunk 1]
```diff
@@ -36,6 +36,11 @@ export const mockMwsFail = new MWS(
   new HttpClient(httpConfig, () => Promise.resolve({ data: '', headers: {} })),
 )
 
-export const parsingError = 'Expected an object, but received a string with value ""'
+export const mockParsingError = new MWS(
+  new HttpClient(httpConfig, () => Promise.resolve({ data: '<xml></xml>', headers: {} })),
+)
+
+export const parsingError = 'Start tag expected.'
+// export const parsingError = 'Expected an object, but received a string with value ""'
```
Comment (moltar) [inline test/utils.ts:44]: Do we still need this comment here?

Comment (justinemmanuelmercado) [reply test/utils.ts:44]: Actually working on this branch now, I'll be removing this comment. Just wanted it there just as a comparison to what it previously was

**Comment by justinemmanuelmercado** [comment]
> So, this is what you were talking about as far as the coverage drop?

Yes but in a way I think it's correct that the coverage dropped. I think how the library tested parsing errors was incorrect we didn't really validate if an XML was valid before having it be decoded, that's why the error being checked before was `'Expected an object, but received a string with value ""'` instead of something like `invalid XML`. 

A better unit test I had in mind is to pass in a valid XML but have it purposefully not match the expected interface. Atleast in this way, FXP catches invalid XML before it gets decoded. I'm in the process of replacing how parsing errors is tested, I'm almost finished and I'll for your review then

**Comment by moltar** [comment]
Ok, sounds great. Yeah, if we can actually test that branch of code, it'd be better of course. I just wasn't sure if maybe the branch was awkward of inconvenient to test. In any case, looking forward! :)

**Comment by justinemmanuelmercado** [comment]
Hey @moltar here's what I had in mind. That way FXP validation gets tested in `http.test.ts` and each method gets each of their `Left: (error) ...` tested (this part)
https://github.com/ScaleLeap/amazon-mws-api-sdk/blob/8bcf3013ef03b4d4880d478781e0e49682f2f714/src/sections/feeds.ts#L219-L221

**Pull request merged**